src/tools/pub/cmd/pub-add.md

```diff
@@ -26,65 +26,151 @@ For example, if 0.13.3 is the latest stable version of the `http` package,
 
-### `--git-url=`_`<git_repo_url>`_
+[dev dependency]: /tools/pub/dependencies#dev-dependencies
+
+```terminal
+$ dart pub add dev:foo           # adds newest compatible stable version of foo
+$ dart pub add dev:foo:^2.0.0    # adds specified constraint of foo
+$ dart pub add foo dev:bar`      # adds regular dependency foo and dev dependency bar simultaneously
+```
+
+_Previously the `-d, --dev` option_:
+
+```terminal
+$ dart pub add --dev foo
+```
+
+### `git` 
+
+Adds a git dependency. 
+
+```terminal
+# dart pub add 'foo{"git":"https://github.com/foo/foo"}'
+```
+
+You can also specify the repository, and the branch or commit, or exact location, within that repository:
+
+```terminal
+# dart pub add 'foo{"git":{"url":"../foo.git","ref":"branch","path":"subdir"}}'
+```
+
+#### `url` {#git-urlgit_repo_url}
 
 Depends on the package in the
 [specified Git repository](/tools/pub/dependencies#git-packages).
 
+```terminal
+# dart pub add 'foo{"git":"https://github.com/foo/foo"}'
+```
+
+_Previously the `--git-url=<git_repo_url>` option_:
+
 ```terminal
 $ dart pub add http --git-url=https://github.com/my/http.git
 ```
 
-### `--git-ref=`_`<branch_or_commit>`_
+#### `ref` {#git-refbranch_or_commit}
+
+With `url`, depends on the specified branch or commit of a Git repo.
 
-With `--git-url`, depends on the specified branch or commit of a Git repo.
+_Previously the `--git-ref=<branch_or_commit>` option_:
 
 ```terminal
 $ dart pub add http --git-url=https://github.com/my/http.git --git-ref=tmpfixes
 ```
 
-### `--git-path=`_`<directory_path>`_
+#### `path` {#git-pathdirectory_path}
+
+With `url`, specifies the location of a package within a Git repo.
+
+_Previously the `--git-path=<directory_path>` option_.
+
+### `hosted` {#hosted-urlpackage_server_url}
+
+Adds a hosted dependency that depends on
+the package server at the specified URL
+
+```terminal
+$ dart pub add 'foo:{"hosted":"my-pub.dev"}'
+```
+
+_Previously the `--hosted-url=<package_server_url>` option_
 
-With `--git-url`, specifies the location of a package within a Git repo.
+### `path` {#pathdirectory_path}
 
-### `--hosted-url=`_`<package_server_url>`_
+Adds path dependency on a locally stored package.
 
-Depends on the package server at the specified URL.
+```terminal 
+$ dart pub add 'foo:{"path":"../foo"}'
+```
 
-### `--path=`_`<directory_path>`_
+_Previously the `--path=<directory_path>` option_.
 
-Depends on a locally stored package.
+### `sdk` {#sdksdk_name}
 
-### `--sdk=`_`<sdk_name>`_
+Adds a package with the specified SDK dependency.
 
-Depends on a package that's shipped with the specified SDK
-(example: `--sdk=flutter`).
+```terminal
+$ dart pub add 'foo:{"sdk":"flutter"}'
+```
+
+_Previously the `--sdk=<sdk_name>` option_:
+
+```terminal
+$ dart pub add foo --sdk=flutter
+```
+
+## Options
+
+For options that apply to all pub commands, see
+[Global options](/tools/pub/cmd#global-options).
+
+{{site.alert.note}}
+  Any specified options will apply to all the packages
```

Comment:
That is why we introduced the new syntax. Maybe it could somehow be highlighted that this note refers to the old syntax?